faith_presentation/src/commonMain/kotlin/net/thechance/mena/faith/presentation/feature/main/MainMapper.kt

```diff
@@ -0,0 +1,123 @@
+import kotlin.time.ExperimentalTime
+import kotlin.time.Instant
+
+object MainMapper {
+    private val PRAYER_ORDER = listOf(
+        PrayerName.FAJR,
+        PrayerName.DHUHR,
+        PrayerName.ASR,
+        PrayerName.MAGHRIB,
+        PrayerName.ISHA
+    )
+
+    @OptIn(ExperimentalTime::class)
+    fun formatTime(instant: Instant): String {
+        val instant = Instant.fromEpochMilliseconds(instant.toEpochMilliseconds())
+        val localDateTime = instant.toLocalDateTime(TimeZone.currentSystemDefault())
+
+        val hour = localDateTime.hour.toString().padStart(2, '0')
+        val minute = localDateTime.minute.toString().padStart(2, '0')
+
+        return "$hour:$minute"
+    }
+
+    fun getPrayerDisplayNameResource(prayerName: PrayerName): StringResource {
+        return when (prayerName) {
+            PrayerName.FAJR -> Res.string.fajr
+            PrayerName.DHUHR -> Res.string.dhuhr
+            PrayerName.ASR -> Res.string.asr
+            PrayerName.MAGHRIB -> Res.string.maghrib
+            PrayerName.ISHA -> Res.string.isha
+            PrayerName.SUNRISE -> Res.string.sunrise
+        }
+    }
+
+    @OptIn(ExperimentalTime::class)
+    fun isAM(instant: Instant): Boolean {
+        val kotlinxInstant = Instant.fromEpochMilliseconds(instant.toEpochMilliseconds())
+        val localDateTime = kotlinxInstant.toLocalDateTime(TimeZone.currentSystemDefault())
+        return localDateTime.hour < 12
+    }
+
+    @OptIn(ExperimentalTime::class)
+    fun List<PrayerTime>.toUi(now: Instant): PrayerTimesUiState {
+        val prayerTimesMap = associateBy { it.name }
+        val prayerTimesUi = PRAYER_ORDER.mapNotNull { prayerTimesMap[it]?.toUi() }
+
+        val currentPrayer = getCurrentPrayer(this, now)
+        val currentPrayerIndex = PRAYER_ORDER.indexOf(currentPrayer)
+
+        return PrayerTimesUiState(
+            prayers = prayerTimesUi,
+            currentPrayerIndex = currentPrayerIndex
+        )
+    }
+
+    @OptIn(ExperimentalTime::class)
+    private fun PrayerTime.toUi(): PrayerUiModel = PrayerUiModel(
+        name = this.name,
+        displayName = getPrayerDisplayNameResource(this.name),
+        time = formatTime(this.time),
+        isAM = isAM(this.time)
+    )
+
+
+    @OptIn(ExperimentalTime::class)
+    private fun getCurrentPrayer(prayerTimes: List<PrayerTime>, now: Instant): PrayerName {
+        val sortedPrayers = prayerTimes
+            .filter { it.name != PrayerName.SUNRISE }
+            .sortedBy { it.time }
+
+        if (now >= sortedPrayers.last().time) {
+            return sortedPrayers.last().name
+        }
+
+        return sortedPrayers.firstOrNull { now < it.time }?.name ?: sortedPrayers.first().name
+    }
+
+    @OptIn(ExperimentalTime::class)
+    fun getSunriseTime(prayerTimes: List<PrayerTime>): String {
+        return prayerTimes
+            .firstOrNull { it.name == PrayerName.SUNRISE }
+            ?.let { formatTime(it.time) }
+            ?: ""
+    }
```

Comment:
```suggestion
    fun getSunriseTime(prayerTimes: List<PrayerTime>): String =
        prayerTimes
            .firstOrNull { it.name == PrayerName.SUNRISE }
            ?.let { formatTime(it.time) } ?: ""
```

And I recommend being in the extension package as an extension package